ports/stm32/boards/ARDUINO_OPTA/mpconfigboard.h

```diff
@@ -0,0 +1,239 @@
+#define MICROPY_HW_BDEV_SPIFLASH_SIZE_BYTES (MICROPY_HW_SPIFLASH_SIZE_BITS / 8)
+#define MICROPY_HW_BDEV_SPIFLASH_EXTENDED (&spi_bdev)
+#endif
+
+// 4 wait states
+#define MICROPY_HW_FLASH_LATENCY    FLASH_LATENCY_2
+
+// UART config
+#define MICROPY_HW_UART3_TX         (pyb_pin_UART3_TX)
+#define MICROPY_HW_UART3_RX         (pyb_pin_UART3_RX)
+#define MICROPY_HW_UART3_RTS        (pyb_pin_UART3_RTS)
+#define MICROPY_HW_UART3_CTS        (pyb_pin_UART3_CTS)
+
+#define MICROPY_HW_UART4_TX         (pyb_pin_BT_TXD)
+#define MICROPY_HW_UART4_RX         (pyb_pin_BT_RXD)
+#define MICROPY_HW_UART4_RTS        (pyb_pin_BT_RTS)
+#define MICROPY_HW_UART4_CTS        (pyb_pin_BT_CTS)
+
+#define MICROPY_HW_UART6_TX         (pyb_pin_UART6_TX)
+#define MICROPY_HW_UART6_RX         (pyb_pin_UART6_RX)
+
+// I2C buses
+#define MICROPY_HW_I2C1_SCL         (pyb_pin_I2C1_SCL)
+#define MICROPY_HW_I2C1_SDA         (pyb_pin_I2C1_SDA)
+
+#define MICROPY_HW_I2C3_SCL         (pyb_pin_I2C3_SCL)
+#define MICROPY_HW_I2C3_SDA         (pyb_pin_I2C3_SDA)
+
+// USRSW is pulled low. Pressing the button makes the input go high.
+#define MICROPY_HW_USRSW_PIN        (pyb_pin_USER_BUTTON)
+#define MICROPY_HW_USRSW_PULL       (GPIO_PULLUP)
+#define MICROPY_HW_USRSW_EXTI_MODE  (GPIO_MODE_IT_FALLING)
+#define MICROPY_HW_USRSW_PRESSED    (0)
+
+// LEDs
+#define MICROPY_HW_LED1             (pyb_pin_LED_SYS_1) // user 1
+#define MICROPY_HW_LED2             (pyb_pin_LED_SYS_2) // user 2
+#define MICROPY_HW_LED_ON(pin)      (mp_hal_pin_low(pin))
+#define MICROPY_HW_LED_OFF(pin)     (mp_hal_pin_high(pin))
+
+// WiFi SDMMC
+#define MICROPY_HW_SDIO_SDMMC       (1)
+#define MICROPY_HW_SDIO_CK          (pyb_pin_WL_SDIO_CLK)
+#define MICROPY_HW_SDIO_CMD         (pyb_pin_WL_SDIO_CMD)
+#define MICROPY_HW_SDIO_D0          (pyb_pin_WL_SDIO_0)
+#define MICROPY_HW_SDIO_D1          (pyb_pin_WL_SDIO_1)
+#define MICROPY_HW_SDIO_D2          (pyb_pin_WL_SDIO_2)
+#define MICROPY_HW_SDIO_D3          (pyb_pin_WL_SDIO_3)
+
+// CYW43 config
+#define CYW43_WIFI_NVRAM_INCLUDE_FILE "wifi_nvram_1dx.h"
+
+// USB config
+#define MICROPY_HW_USB_FS           (1)
+
+#define MICROPY_HW_USB_CDC_RX_DATA_SIZE     (1024)
+#define MICROPY_HW_USB_CDC_TX_DATA_SIZE     (1024)
+#define MICROPY_HW_USB_CDC_1200BPS_TOUCH    (1)
+
+// Bluetooth config
+#define MICROPY_HW_BLE_UART_ID              (PYB_UART_4)
+#define MICROPY_HW_BLE_UART_BAUDRATE        (115200)
+#define MICROPY_HW_BLE_UART_BAUDRATE_SECONDARY (3000000)
+
+// Ethernet via RMII
+#define MICROPY_HW_ETH_MDC          (pin_C1)
+#define MICROPY_HW_ETH_MDIO         (pin_A2)
+#define MICROPY_HW_ETH_RMII_REF_CLK (pin_A1)
+#define MICROPY_HW_ETH_RMII_CRS_DV  (pin_A7)
+#define MICROPY_HW_ETH_RMII_RXD0    (pin_C4)
+#define MICROPY_HW_ETH_RMII_RXD1    (pin_C5)
+#define MICROPY_HW_ETH_RMII_TX_EN   (pin_G11)
+#define MICROPY_HW_ETH_RMII_TXD0    (pin_G13)
+#define MICROPY_HW_ETH_RMII_TXD1    (pin_G12)
+
+#define MICROPY_HW_USB_VID                      0x2341
```

Comment:
Please move these settings up to the same block as `MICROPY_HW_USB_FS` is defined.

Reply:
I moved this line down to better match the other Arduino board (they all have this split).

Reply:
OK.